kernel/idle.c

```diff
@@ -152,6 +157,24 @@ void idle(void *unused1, void *unused2, void *unused3)
 #endif
 
 	while (true) {
+		/* Lock interrupts to atomically check if to_abort is non-NULL,
+		 * and if so clear it
+		 */
+		int key = arch_irq_lock();
+		struct k_thread *to_abort = cpu->pending_abort;
+
+		if (to_abort) {
+			cpu->pending_abort = NULL;
+			arch_irq_unlock(key);
+
+			LOG_DBG("idle %p aborting thread %p",
+				_current, to_abort);
+
+			z_thread_single_abort(to_abort);
+			z_reschedule_unlocked();
```

Comment:
This bit needs a comment, I think.  Normally, the fact that the idle thread is running at all means by definition that no other higher priority threads are available.  So a reschedule here really should be a noop.

But it's not now, because the thread abort case means that the idle thread can also run as the "highest" priority too, in which case it can't ignore the reschedule because it would start spinning until the next interrupt.

That's subtle and surprising, so we should call it out here so no one comes by and tries to remove the reschedule to be helpful.

Reply:
will do